csm_api_client/k8s.py

```diff
@@ -0,0 +1,60 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+"""
+Utility functions for interacting with Kubernetes.
+"""
+
+import warnings
+
+from kubernetes.client import CoreV1Api
+from kubernetes.config import load_kube_config
+from kubernetes.config.config_exception import ConfigException
+# YAMLLoadWarning is not defined in types-PyYAML.
+from yaml import YAMLLoadWarning  # type: ignore[attr-defined]
+
+
+def load_kube_api() -> CoreV1Api:
+    """Get a Kubernetes CoreV1Api object.
+
+    This helper function loads the kube config and then instantiates
+    an API object.
+
+    Returns:
+        CoreV1Api: the API object from the kubernetes library.
```

Comment:
I gave it some thought, but for the time being I deferred removing the docstring comments. I think it would be best to remove them, though I was more focused on getting the hints, unittests, and packaging up and running at the time. Leaving them in is likely a long-term liability.